### PR TITLE
Enable cancel-in-progress in ci

### PR DESCRIPTION
**Pull request by yamatcha**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -6,6 +6,10 @@ on:
     branches:
       - 'main'
 
+concurrency:
+  group: ${{ github.head_ref || github.run_id }}
```
Comment (ymmt2005) [inline .github/workflows/ci.yaml:10]: Just a suggestion
```suggestion
  group: ${{ github.head_ref || github.run_id }}-ci
```

Comment (ymmt2005) [reply .github/workflows/ci.yaml]: In the official reference, they actually use `${{ github.workflow }}` to specify the workflow.

Comment (yamatcha) [reply .github/workflows/ci.yaml]: Thank you. I've fixed as you suggested.

Comment (ymmt2005) [reply .github/workflows/ci.yaml]: I mean, if we use `${{ github.workflow }}`, we don't need to add `-e2e` or `-ci`.

**Review by ymmt2005 — APPROVED** [review]
LGTM

**Pull request merged**